app/token.py

```diff
@@ -21,17 +19,46 @@ def user_role():
 
 def admin_required(f):
     def decorator():
-        record = user_role()
-        if record['role'] != "ADMIN":
+        payload = request.headers["Authorization"]
```

Comment:
to check the token we are writing the same code in all 3 decorators make a function instead for this check.